framework/auth/core.py

```diff
@@ -42,17 +43,19 @@
 logger = logging.getLogger(__name__)
 
 
-# Hide implementation of token generation
-def generate_confirm_token():
-    return security.random_string(30)
-
-
-def generate_claim_token():
+# verification key v1
+def generate_verification_key():
     return security.random_string(30)
 
 
-def generate_verification_key():
-    return security.random_string(30)
+# verification key v2, expires in 30 minutes
+def generate_verification_key_v2():
```

Comment:
Not a blocker, but if the main purpose of this PR is to normalize (as much as possible) user / password confirmation flow, and you're already de-duplicating token generation, why not have a single helper function for this? e.g.
```python
#### defaults.py
EXPIRATION_TIME_DICT = {
    'verify': 30,
    'confirm': 24 * 60,  # 24h in minutes
    'claim': 7 * 24 * 60  # 7d in minutes
}

### here
# Define constants (VERIFY = 'verify', etc.), to be imported elsewhere
def generate_verification_key(verification_type=None):
    token = security.random_string(30)
    if not verification_type:  # Assume v1
        return token
    expires = dt.datetime.utcnow() + dt.timedelta(minutes=settings.EXPIRATION_TIME_DICT[verification_type])
    return {
        'token': token,
        'expires': expires
    }
```